python/lsst/cp/pipe/defects.py

```diff
@@ -367,27 +393,91 @@ def _writeData(self, dataRef, defects):
         defects : `lsst.meas.algorithms.Defect`
             The defects to be written.
         """
-        filename = dataRef.getUri(write=True)  # does not guarantee that full path exists
-        dirname = os.path.dirname(filename)
-        if not os.path.exists(dirname):
-            os.makedirs(dirname)
+        date = str(dafBase.DateTime(midTime, dafBase.DateTime.MJD).toPython())
+        date = date.replace(" ", "T")  # original has a space in the middle
```

Comment:
Can this be done as `date = dafBase.DateTime(midTime, dafBase.DateTime.MJD).toPython().isoformat()`, so the formatting options are done on the datetime object instead of messing with the string?

Reply:
Yup - done. Much nicer, thanks.